arangod/Aql/SimpleModifier.cpp

```diff
@@ -192,8 +195,53 @@ ExecutionState SimpleModifier<ModifierCompletion, Enable>::transact(
   // operations blocking as in previous versions of ArangoDB.
   // TODO: fix this and make it truly non-blocking (requires to
   // fix some lifecycle issues for AQL queries first).
-  result.wait();
 
+  // If we simply wait, it can happen that we get into a blockage in which
+  // all threads wait in the same place here and none can make progress,
+  // since the scheduler is full. This means we must detach the thread
+  // after some time. To avoid that all are detaching at the same time,
+  // we choose a random timeout for the detaching. But first we spin a
+  // while to avoid delays:
+  if (!result.isReady()) {
+    auto const spinTime = std::chrono::microseconds(10'000);
+    auto start = std::chrono::steady_clock::now();
+    while (!result.isReady() &&
+           std::chrono::steady_clock::now() - start < spinTime) {
+      basics::cpu_relax();
+    }
+    if (!result.isReady()) {
+      auto const detachTime = std::chrono::microseconds(
+          RandomGenerator::interval(uint32_t(100)) * 100'000);
```

Comment:
I think that this can lead to a detachTime of `0` in the worst case.
If the rng returns `0`, then the effective wait time is only 10ms (because of the 10ms spinning before).
I think we should at a minimum offset value of at least 1 second here, maybe even more, e.g.
```
      auto const detachTime = std::chrono::microseconds(
          1'000'000 + RandomGenerator::interval(uint32_t(100)) * 100'000);

Reply:
Done with 1s.